### PR TITLE
Some Basic Improvements 

### PR DESCRIPTION
**Pull request by itmard**
+ Update gitignore file ( Using standard python gitignore file from github: https://github.com/github/gitignore/blob/master/Python.gitignore)
+ Getting Application Secret Key and other secrets ( RECAPTCHA_SECRET_KEY and POSTMARK_API_TOKEN ) from Environment variables.
+ Create a utils file for _Web_ application and move reusable codes and utils to this file
+  Using Django crypto get_random_string in register view to generate random strings ( far better and faster than using _string_ and _random_. Also this not predictable by an attacker) 
You can change allowed_chars for the random string seed.
+ Optimize imports in views
+ Using relative imports inside applications ( Using relative imports for inside app imports based on Django coding style:
https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#imports )
🐼🐼🐼

### CODE REVIEW

**Comment by itmard** [comment]
@jadijadi Sorry this merge request has some conflicts on views file. I can't fix them because I don't know the logic behind this application. Please review and fix them before merging this merge request.

File: bestoon/settings.py [hunk 1]
```diff
@@ -15,19 +15,21 @@
...
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/1.10/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'ees#cr8y%!0o=%8r2wbxcgcz0u8s!&91=b2f@a_c%&4*c&ey&a'
+# You should keep the secret key used in production secret and do not share it
+# on Github with us.
+# Getting Application Secret Key from Environment variables
+# You should set GENERAL_SECRET_KEY=You_Secret_Key
+SECRET_KEY = os.environ.get('GENERAL_SECRET_KEY')
```
Comment (arashrasoulzadeh) [inline bestoon/settings.py]: maybe 
+SECRET_KEY = os.environ.get('bestoon_GENERAL_SECRET_KEY')

File: bestoon/settings.py [hunk 2]
```diff
@@ -114,8 +113,11 @@
...
 USE_TZ = True
 
-
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/1.10/howto/static-files/
 
 STATIC_URL = '/static/'
+
+# Secrets
+RECAPTCHA_SECRET_KEY = os.environ.get('RECAPTCHA_SECRET_KEY')
```
Comment (arashrasoulzadeh) [inline bestoon/settings.py]: +RECAPTCHA_SECRET_KEY = os.environ.get('bestoon_RECAPTCHA_SECRET_KEY')

Comment (ShalbafZadeh) [inline bestoon/settings.py]: @arashrasoulzadeh i think it's a convention for environment variables to be UPPERCASE 
 i suggest 
```
+RECAPTCHA_SECRET_KEY = os.environ.get('BESTOON_RECAPTCHA_SECRET_KEY')
```

**Pull request merged**